src/utils/error-handler.ts

```diff
@@ -1,82 +1,29 @@
 import {FastifyError, FastifyRequest, FastifyReply} from 'fastify';
+import {ErrorDataFormatter, ErrorResponseFormatter} from './error-formatters';
 
 /**
  * Creates a structured validation error handler for Fastify that logs validation
  * errors with detailed context and returns consistent error responses.
- * 
+ *
  * This handler specifically catches schema validation errors (statusCode 400 with validation property)
  * and logs them with structured data suitable for GCP logging and monitoring.
- * 
+ *
  * @returns A Fastify error handler function
  */
 export function errorHandler() {
     return (error: FastifyError, request: FastifyRequest, reply: FastifyReply) => {
-        // Only handle validation errors specifically
-        if (error.statusCode === 400 && error.validation) {
-            // Log validation error details with structured format
-            reply.log.warn({
-                err: {
-                    message: error.message,
-                    name: error.name,
-                    code: error.code,
-                    stack: error.stack,
-                    validationContext: error.validationContext, // 'query', 'headers', 'body'
-                    validation: error.validation // Array of validation errors
-                },
-                httpRequest: {
-                    requestMethod: request.method,
-                    requestUrl: request.url,
-                    userAgent: request.headers['user-agent'],
-                    remoteIp: request.ip,
-                    referer: request.headers.referer,
-                    status: 400
-                },
-                headers: {
-                    'content-type': request.headers['content-type'],
-                    'x-site-uuid': request.headers['x-site-uuid'],
-                    'user-agent': request.headers['user-agent'],
-                    referer: request.headers.referer
-                },
-                query: request.query,
-                requestBody: request.body,
-                type: 'validation_error'
-            }, 'Schema validation failed');
-            
-            // Return structured error response to client
-            return reply.status(400).send({
-                error: 'Bad Request',
-                message: error.message,
-                statusCode: 400,
-                validation: error.validation
-            });
+        const handleValidationErrors = error.statusCode === 400 && error.validation;
```

Comment:
nit: this will only ever handle one error at a time, so I think this should be singular. Same thing where it's used below this, of course.

```suggestion
        const handleValidationError = error.statusCode === 400 && error.validation;
```

Reply:
ah I see, updated 👍